src/types/interface.ts

```diff
@@ -277,11 +344,101 @@ export class InterfaceType<Props extends Properties, ResultType>
             ...otherPropsInfo,
         };
         return createType(
-            new InterfaceType(propsInfo as PropertiesInfo<Omit<Props, keyof OtherProps> & OtherProps>, { ...this.options, name }),
+            new InterfaceType(propsInfo as PropertiesInfo<Omit<Props, keyof OtherProps> & OtherProps>, {
+                ...this.options,
+                name: this._deriveName(name, s => s),
+            }),
             { customValidators: { configurable: true, value: customValidators } },
         );
     }
 
+    /**
+     * Create a new type that consists only of the mentioned properties similar to the builtin `Pick` type.
+     */
+    pick<const Key extends keyof Props & keyof ResultType & string>(
+        ...args: [keys: OneOrMore<Key>] | [name: string, keys: OneOrMore<Key>] | [options: InterfacePickOptions, keys: OneOrMore<Key>]
+    ): PickType<Props, ResultType, Key> {
+        const [opts, keys] = decodeOptionalOptions<InterfacePickOptions, OneOrMore<Key>>(args);
+        return this._pick(opts, keys, 'pick', name => `Pick<${name}, ${keys.map(stringStringify).join(' | ')}>`);
+    }
+
+    /**
+     * Create a new type that consists of all properties of the base type, except those mentioned, similar to the builtin `Omit` type.
+     */
+    omit<const Key extends keyof Props & keyof ResultType & string>(
+        ...args: [keys: OneOrMore<Key>] | [name: string, keys: OneOrMore<Key>] | [options: InterfacePickOptions, keys: OneOrMore<Key>]
+    ): PickType<Props, ResultType, Exclude<keyof Props & keyof ResultType & string, Key>> {
+        const [opts, keys] = decodeOptionalOptions<InterfacePickOptions, OneOrMore<Key>>(args);
+        const keysToOmit = new Set<string>(keys);
+        const pickKeys = this.keys.filter(key => !keysToOmit.delete(key));
+        // Trick to reuse the error message about keys that were not found:
+        pickKeys.push(...keysToOmit);
```

Comment:
So of both 'Operations mentions one or more keys that do not exist on the base type' and 'All properties omitted' errors can be thrown, we pick the first one? 🤔  It seems that omitting keys that don't exist may be the biggest 'red flag' if both are true. But it could use a (small) comment about the reasoning behind this.